digits/frameworks/__init__.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2014-2015, NVIDIA CORPORATION.  All rights reserved.
+
+from framework import Framework
+from caffe_framework import CaffeFramework
+from digits.config import config_value
+
+#
+#  create framework instances
+#
+
+# caffe is mandatory
+caffe = CaffeFramework()
+
+#
+#  utility functions
+#
+
+def get_frameworks():
+    """
+    return list of all available framework instances
+    there may be more than one instance per framework class
+    """
+    frameworks = [caffe]
+    return frameworks
+
+def get_framework_by_id(id):
```

Comment:
It's bad form to use `id` as a variable name since it's also a python builtin function.

https://docs.python.org/2/library/functions.html#id